src/build.py

```diff
@@ -611,9 +611,16 @@ def __init__(self, include=None, exclude=None):
 
   def Apply(self, targets):
     """ Return the filtered list of targets. """
+    all_names = [t.name for t in targets]
     if self.include is not None:
+      missing_includes = [i for i in self.include if i not in all_names]
+      if missing_includes:
+        raise Exception('Invalid step name(s): ' + str(missing_includes))
       return [t for t in targets if t.name in self.include]
     if self.exclude:
+      missing_excludes = [i for i in self.include if i not in all_names]
+      if missing_excludes:
+        raise Exception('Invalid step name(s): ' + str(missing_excludes))
```

Comment:
Not too fond of the duplication, though my suggestion feels kinda icky as well.

Can hoist this out with something like
```python
specified_list = self.include or self.exclude or []
missing = # filter specified_list
if missing: # exception
```

Not sure if that's actually more readable, so leaving it might be fine.